src/metabase/api/collection.clj

```diff
@@ -428,23 +431,40 @@
 
 (api/defendpoint PUT "/:id"
   "Modify an existing Collection, including archiving or unarchiving it, or moving it."
-  [id, :as {{:keys [name color description archived parent_id], :as collection-updates} :body}]
-  {name        (s/maybe su/NonBlankString)
-   color       (s/maybe collection/hex-color-regex)
-   description (s/maybe su/NonBlankString)
-   archived    (s/maybe s/Bool)
-   parent_id   (s/maybe su/IntGreaterThanZero)}
+  [id, :as {{:keys [name color description archived parent_id type type_sub_tree], :as collection-updates} :body}]
+  {name          (s/maybe su/NonBlankString)
+   color         (s/maybe collection/hex-color-regex)
+   description   (s/maybe su/NonBlankString)
+   archived      (s/maybe s/Bool)
+   parent_id     (s/maybe su/IntGreaterThanZero)
+   type          collection/Type
+   type_sub_tree (s/maybe s/Bool)}
```

Comment:
I find this parameter name a little opaque... how about something like `update_descendant_types` or `update_subcollection_types`? "Hey API, please `update_subcollection_types` for me" seems more obvious than "Hey API, please `type_sub_tree` for me" to me. Just my 2 cents

Reply:
that's way better. I was using `type` as a verb, like "type the sub tree" but it didn't feel right to me.